vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
`I'm not sure that CAS works unless you know the last observed revision, not just value`  - See the usages of this API. Its intent is to allow EnableService or DisableService only once per sandbox. The underlying type being worked on is a boolean. Its a low level primitive. Its not a CAS operation on a type that can take multiple values where you need a revision number per operation (ie: typical CAS semantics provided by Consul etc.)